frontend_server/lib/server.dart

```diff
@@ -0,0 +1,160 @@
+library frontend_server;
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:front_end/compilation_message.dart';
+
+import 'package:front_end/incremental_kernel_generator.dart';
+import 'package:front_end/compiler_options.dart';
+import 'package:front_end/kernel_generator.dart';
+import 'package:kernel/ast.dart';
+import 'package:kernel/kernel.dart';
+import 'package:kernel/target/flutter.dart';
+import 'package:kernel/target/flutter_fasta.dart';
+import 'package:kernel/target/targets.dart';
+import 'package:usage/uuid/uuid.dart';
+
+ArgParser argParser = new ArgParser(allowTrailingOptions: true)
+  ..addFlag('train',
+      help: 'Run through sample command line to produce snapshot',
+      negatable: false)
+  ..addFlag('incremental',
+      help: 'Run compiler in incremental mode', defaultsTo: false)
+  ..addOption('sdk-root',
+      help: 'Path to sdk root',
+      defaultsTo: '../../out/android_debug/flutter_patched_sdk')
+  ..addOption('platform-kernel-dill',
+      help: 'Path to platform kernel dill file',
+      defaultsTo: '../../out/android_debug/flutter_patched_sdk/platform.dill');
+
+String usage = '''
+Usage: server [options] [input.dart]
+
+If input dart source code is provided on the command line, then the server 
+compiles it, generates dill file and exits.
+If no input dart source is provided on the command line, server waits for 
+instructions from stdin.
+
+Instructions:
+- compile <input.dart>
+- recompile <boundary-key>
+<invalidate-file1.dart>
```

Comment:
nit: consider using the example `<path/to/invalid/file1.dart>` instead of `invalidate-file` (mostly using the word "invalidate" made me think that you required the "invalidate" command as a prefix, as in:
```
recompile KEY
invalidate file1.dart
invalidate path-to-file2.dart
KEY
```